sdks/go/pkg/beam/core/runtime/graphx/coder.go

```diff
@@ -248,7 +248,9 @@ func (b *CoderUnmarshaller) makeCoder(id string, c *pipepb.Coder) (*coder.Coder,
 		case urnIntervalWindow:
 			// If interval window in a KV, this may be a mapping function.
 			// Special case since windows are not normally used directly as FullValues.
-			return coder.NewIntervalWindowCoder(), nil
+			value := coder.NewIntervalWindowCoder()
+			t := typex.New(root, key.T, value.T)
+			return &coder.Coder{Kind: kind, T: t, Components: []*coder.Coder{key, value}}, nil
```

Comment:
Ah didn't see this. Basically, we should only need to put the `case urnIntervalWindow: return coder.NewIntervalWindowCoder(), nil` into the outer switch (probably just above KV, since it's easier to read), and not explicitly handle it under the KV coder case.

Reply:
Ah, so there is already a case in the switch for that [here](https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/core/runtime/graphx/coder.go#L388-L393), and it talks about special handling for tests. I was hesitant to do anything with that before review. It also looks like there's other behavior that might depend on that being a `Window` coder instead of this new kind.

Reply:
As the person who put that in, I'd rather inconvenience the tests than the production code. Please feel free to change them up. (sorry for the churn).
That was before I was aware that there were uses for runner sending coders around like that, and we just wanted to validate the coders per the standard_coders.yaml tests.

Those uses are handled here usually: https://github.com/camphillips22/beam/tree/c83b42f7547a5fd57e2f9a90d7b613c512eb088c/sdks/go/test/regression/coders/fromyaml